src/Microsoft.ML.CpuMath/SseIntrinsics.cs

```diff
@@ -285,54 +285,164 @@ public static unsafe void MatMul(float[] mat, float[] src, float[] dst, int crow
         }
 
         // Partial sparse source vector.
-        public static unsafe void MatMulPA(AlignedArray mat, int[] rgposSrc, AlignedArray src,
-                                        int posMin, int iposMin, int iposEnd, AlignedArray dst, int crow, int ccol)
+        public static unsafe void MatMulP(AlignedArray mat, int[] rgposSrc, AlignedArray src,
+                                int posMin, int iposMin, int iposEnd, AlignedArray dst, int crow, int ccol)
         {
-            Contracts.Assert(HasCompatibleAlignment(mat));
-            Contracts.Assert(HasCompatibleAlignment(src));
-            Contracts.Assert(HasCompatibleAlignment(dst));
+            Contracts.Assert(crow % 4 == 0);
```

Comment:
I think these asserts should move into the method below (the one taking in `float[] mat, int[] rgposSrc`). That method is the one that requires `ccol` to be a multiple of 4.